mbuild/recipes/silica_interface.py

```diff
@@ -0,0 +1,151 @@
+        super(SilicaInterface, self).__init__()
+
+        O_buffer = 0.275
+
+        self._cleave_interface(bulk_silica, tile_x, tile_y, thickness, O_buffer)
+        self.generate_bonds(name_a='Si', name_b='O', dmin=0.0, dmax=0.20419)
+        self._strip_stray_atoms()
+        self._bridge_dangling_Os(oh_density, thickness)
+        self._identify_surface_sites(thickness)
+        self._adjust_stoichiometry(O_buffer)
+
+    def _cleave_interface(self, bulk_silica, tile_x, tile_y, thickness, O_buffer):
+        """ Carve interface from bulk silica, include a buffer of O's above and
+            below the surface to ensure the interface is coated.
+        """ 
+
+        tile_z = int(ceil((thickness+(2*O_buffer))/bulk_silica.periodicity[2]))
+        bulk = mb.TiledCompound(bulk_silica, n_tiles=(tile_x, tile_y, tile_z))
+
+        interface = mb.Compound(periodicity=(bulk.periodicity[0],bulk.periodicity[1],
+                                             0.0))
+        for i, particle in enumerate(bulk.particles()):
+            if particle.name == 'Si' and particle.pos[2] < (thickness+O_buffer) and particle.pos[2] > O_buffer:
+                temp_particle = mb.Compound(name=particle.name, pos=particle.pos)
+                interface.add(temp_particle, particle.name+"_{}".format(i))
+            elif particle.name == 'O' and particle.pos[2] < (thickness+2.*(O_buffer)):
+                temp_particle = mb.Compound(name=particle.name, pos=particle.pos)
+                interface.add(temp_particle, particle.name+"_{}".format(i))
+        self.add(interface)
+
+    def _strip_stray_atoms(self):    
+        """ Remove stray atoms and surface pieces """
+
+        for i,atom in enumerate(list(self.particles())):
+            if len(self.bond_graph.neighbors(atom)) == 0:
+                self.remove(atom)
+
+        for atom in list(self.particles()):
+            if atom not in max(nx.connected_components(self.bond_graph), key=len):
+                self.remove(atom)
+
+    def _bridge_dangling_Os(self, oh_density, thickness):
+        """ Create Si-O-Si bridges on the surface to yield the desired
+            density of reactive surface sites
+        """        
+
+        area = self.periodicity[0] * self.periodicity[1]
+        target = int(oh_density * area)
+
+        dangling_Os = []
+        for atom in list(self.particles()):
+            if len(self.bond_graph.neighbors(atom)) == 1:
+                if atom.name == 'O' and atom.pos[2] > thickness:
+                    dangling_Os.append(atom)
+
+        n_bridges = int((len(dangling_Os) - target)/2.)
+
+        print("Initial density of surface sites:\t{}".format(len(dangling_Os)/area))
+
+        for i in range(n_bridges):
+            bridged = False
+            while not bridged:
+                j = randint(0, len(dangling_Os)-1)
+                O1 = dangling_Os[j]
+                assert(len(self.bond_graph.neighbors(O1)) == 1)
```

Comment:
These assert statements should probably be moved to an appropriate unit test since it should be impossible to get a length other than 1 based on how you construct the list.